src/main/kotlin/org/jetbrains/intellij/platform/gradle/argumentProviders/SandboxArgumentProvider.kt

```diff
@@ -54,10 +57,36 @@ class SandboxArgumentProvider(
     override fun asArguments() = listOfNotNull(
         sandboxConfigDirectory.ifExists { "-Didea.config.path=$it" },
         sandboxSystemDirectory.ifExists { "-Didea.system.path=$it" },
-        sandboxLogDirectory.ifExists { "-Didea.log.path=$it" },
+        sandboxLogDirectory.ifExists { "-Didea.log.path=$it" }
+    ) + computePluginPathProperties()
+
+    protected open fun computePluginPathProperties() = listOfNotNull(
         sandboxPluginsDirectory.ifExists { "-Didea.plugins.path=$it" },
         pluginPath?.let { "-Dplugin.path=$it" },
     )
 
-    private fun <T> DirectoryProperty.ifExists(block: (Path) -> T) = orNull?.asPath?.takeIf { it.exists() }?.run(block)
+    protected fun <T> DirectoryProperty.ifExists(block: (Path) -> T) = orNull?.asPath?.takeIf { it.exists() }?.run(block)
 }
+
+class SandboxArgumentProviderSplitModeAware(
```

Comment:
I would prefer to avoid creating this subclass, but I don't know how to pass 'always false' property correctly when creating an instance where `splitMode` property is not available.